pkg/webhook/ingress/types.go

```diff
@@ -44,6 +47,24 @@ func (n NetworkingV1) IngressClass() (res *string) {
 	return
 }
 
+func (n NetworkingV1) SetIngressClass(ingressClassName string) {
+	if n.Spec.IngressClassName == nil {
+		if a := n.GetAnnotations(); a != nil {
+			if _, ok := a[annotationName]; ok {
+				a[annotationName] = ingressClassName
+
+				return
+			}
+		}
+	}
+	// Assign in case the IngressClassName property was not set
+	if version, _ := utils.GetK8sVersion(); version.Major() == 1 && version.Minor() < 18 {
+		n.Annotations[annotationName] = ingressClassName
+	} else {
+		n.Spec.IngressClassName = &ingressClassName
+	}
```

Comment:
There's no need for this check since this API is served by the Kubernetes API Server that would reject any attempt against an unserver API group.

With that said, for the newer type we should avoid using the annotation that has been marked as deprecated: this strategy is for the API group networking.k8s.io/v1` and the field `IngressClassName` should be used.